src/main/kotlin/ar/edu/unahur/obj2/vendedores/Vendedor.kt

```diff
@@ -28,24 +28,78 @@ abstract class Vendedor {
   fun otrasCertificaciones() = certificaciones.count { !it.esDeProducto }
 
   fun puntajeCertificaciones() = certificaciones.sumBy { c -> c.puntaje }
+
+  /*puntos*/
+  abstract fun esInfluyente(): Boolean
+
+  fun esGenerico() : Boolean {
+    return this.otrasCertificaciones() >= 1
+  }
 }
 
 // En los parámetros, es obligatorio poner el tipo
 class VendedorFijo(val ciudadOrigen: Ciudad) : Vendedor() {
   override fun puedeTrabajarEn(ciudad: Ciudad): Boolean {
     return ciudad == ciudadOrigen
   }
+
+  override fun esInfluyente(): Boolean {
+    return false
+  }
 }
 
 // A este tipo de List no se le pueden agregar elementos una vez definida
 class Viajante(val provinciasHabilitadas: List<Provincia>) : Vendedor() {
   override fun puedeTrabajarEn(ciudad: Ciudad): Boolean {
     return provinciasHabilitadas.contains(ciudad.provincia)
   }
+  override fun esInfluyente():Boolean {
+    return provinciasHabilitadas.sumBy { p -> p.poblacion } >= 10000000
+  }
 }
 
 class ComercioCorresponsal(val ciudades: List<Ciudad>) : Vendedor() {
+
   override fun puedeTrabajarEn(ciudad: Ciudad): Boolean {
     return ciudades.contains(ciudad)
   }
+  fun cantidadCiudadesConSucursal() : Int { return ciudades.size }
+  fun provinciasConSucursal() : Set<Provincia> { return ciudades.map { c -> c.provincia }.toSet()}
+  fun cantidadProvinciasConSucursal() : Int { return this.provinciasConSucursal().size}
+
+  override fun esInfluyente():Boolean {
+    return cantidadCiudadesConSucursal() >= 5 || cantidadProvinciasConSucursal() >= 3
+  }
 }
+
+class CentroDeDistribucion (val ubicacion: Ciudad){
+  val vendedores = mutableListOf<Vendedor>()
+
+  fun hayVendedor(vendedor:Vendedor): Boolean {
+    return this.vendedores.contains(vendedor)
+  }
+
+  fun agregarVendedor(vendedor: Vendedor) {
+    if (!hayVendedor(vendedor)) {
+      vendedores.add(vendedor)
+    } else {
+      throw error("El vendedor ya está registrado")
+    }
```

Comment:
Si se ejecuta la rama verdadera del `if`, el método explota y nunca llega al `else`. Podés entonces invertirlo, te quedaría así:

```suggestion
    if (hayVendedor(vendedor)) {
      error("El vendedor ya está registrado")
    } 
    vendedores.add(vendedor)
```